### PR TITLE
chore: add polymorphDafny and polymorphDotnet tasks to Smithy Build Plugin-based SQS TestModel

### PR DESCRIPTION
**Pull request by kessplas**
*Issue #, if available:* https://github.com/smithy-lang/smithy-dafny/issues/356 (related)

*Description of changes:* This PR allows the Smithy Build Plugin-based SQS TestModel to be built and structured similarly to the Makefile/CLI based models used elsewhere. This enables some Smithy features which are not supported by the CLI, such as [Projections](https://smithy.io/2.0/guides/smithy-build-json.html#projections). 

Instead of running `make polymorph_dafny` and `make polymorph_dotnet`, run `./gradlew polymorphDafny` and `./gradlew polymorphDotnet` to generate types for the respective languages. A Makefile is included, as it is needed to run `make transpile_net` and `make test_net`, which work as expected when the above commands are executed. 

NOTE: These changes are not tested in CI. That may be done as part of this PR, or later, depending on what reviewers prefer. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: TestModels/aws-sdks/sqs/smithy-build.json [hunk 1]
```diff
@@ -1,5 +1,23 @@
 {
   "version": "1.0",
+  "projections": {
```
Comment (robin-aws) [inline TestModels/aws-sdks/sqs/smithy-build.json:3]: Could we make this `listQueuesOnly` and actually hook it up as the source in the Gradle build via `projectionName = "listQueuesOnly"` as above? That way we're actually testing the workflow we're claiming to work now, and `TestComAmazonawsSqs.dfy` should still work AFAICT.

Comment (kessplas) [reply TestModels/aws-sdks/sqs/smithy-build.json:3]: Yeah, good call!

Comment (robin-aws) [reply TestModels/aws-sdks/sqs/smithy-build.json:3]: Thanks! Only nit is that "listQueuesOnly" is now technically "listQueuesOnlyExceptForAtLeastOneOtherOneBecauseOf#439" :) But it doesn't look like `smithy-build.json` supports comments. Perhaps we could add a comment to the README of this test model?

**Review by robin-aws — DISMISSED** [review]

File: TestModels/aws-sdks/sqs/smithy-build.json (hunk 1 above)
Comment (kessplas) [reply TestModels/aws-sdks/sqs/smithy-build.json:3]: good point, I'll do both. 
(JSON in general doesn't support comments, it's kind of a bummer)

**Review by robin-aws — APPROVED** [review]

**Pull request merged**